qa/rpc-tests/wallet_mergetoaddress_2.py

```diff
@@ -0,0 +1,341 @@
+        result = self.z_mergetoaddress_tz2t(4, 4)
+        assert_equal(result[self.op_result_k][self.mergingUTXOs_k], 4)
+        assert_equal(result[self.op_result_k][self.mergingNotes_k], 4)
+        assert_equal(result[self.op_result_k][self.mergingTransparentValue_k] + result[self.op_result_k][self.mergingShieldedValue_k], result[self.balance_delta_k])
+        #number of joinsplits analogous to t2t + z2t
+        assert_equal(result[self.joinsplit_len_k], 3)
+        # ---------- TZ to T ----------
+
+
+        # ---------- TZ to Z ----------
+        #move 3 utxos from n1 to n0
+        self.send_transparent(3)
+        #merge up to 3 utxos (available more than 3) and up to 3 notes (available more than 3) from n0 to n2
+        result = self.z_mergetoaddress_tz2z(3, 3)
+        assert_equal(result[self.op_result_k][self.mergingUTXOs_k], 3)
+        assert_equal(result[self.op_result_k][self.mergingNotes_k], 3)
+        assert_equal(result[self.op_result_k][self.mergingTransparentValue_k] + result[self.op_result_k][self.mergingShieldedValue_k], result[self.balance_delta_k])
+        #number of joinsplits analogous to t2z + z2z, but consider the joinsplit required by t2z is actually collpased in the final z2z joinsplit
+        assert_equal(result[self.joinsplit_len_k], 2)
+
+        #move 4 utxos from n1 to n0
+        self.send_transparent(4)
+        #merge up to 4 utxos (available more than 4) and up to 4 notes (available more than 4) from n0 to n2
+        result = self.z_mergetoaddress_tz2z(4, 4)
+        assert_equal(result[self.op_result_k][self.mergingUTXOs_k], 4)
+        assert_equal(result[self.op_result_k][self.mergingNotes_k], 4)
+        assert_equal(result[self.op_result_k][self.mergingTransparentValue_k] + result[self.op_result_k][self.mergingShieldedValue_k], result[self.balance_delta_k])
+        #number of joinsplits analogous to t2z + z2z, but consider the joinsplit required by t2z is actually collpased in the final z2z joinsplit
+        assert_equal(result[self.joinsplit_len_k], 3)
+        # ---------- TZ to Z ----------
+
+
+        # ---------- big size ----------
+        #merge up to 60 notes (available more than 60, but transaction size limit admits up to 58) from n0 to n2
+        result = self.z_mergetoaddress_z2z(60)
+        assert_equal(result[self.op_result_k][self.mergingUTXOs_k], 0)
+        assert_equal(result[self.op_result_k][self.mergingNotes_k], 58)
+        assert_equal(result[self.op_result_k][self.mergingTransparentValue_k], 0)
+        assert_equal(result[self.op_result_k][self.mergingShieldedValue_k], result[self.balance_delta_k])
+        #57 joinsplits are required because first two shielded inputs are merged in a shielded output,
+        #which is then set as shielded input (change) in the subsequent joinsplit and paired with another shielded input
+        #until the final shielded output is obtained
+        assert_equal(result[self.joinsplit_len_k], 57)
```

Comment:
I guess that this test is important to check the behavior of the implemented fix. I would expect that this test was previously failing due to a wrong estimation of the transaction size, while after the fix works as expected. Am I right?

If so, I'm not sure if all the previous tests and the last one are relevant to this PR.

Reply:
Removed other tests, moved tz2t test to `wallet_mergetoaddress.py`